src/components/container/IntradayHeartRateChart/IntradayHeartRateChart.tsx

```diff
@@ -0,0 +1,133 @@
+                    data = MissingMidDayData;
+                    break;
+            }
+
+            transformToChartData(data);
+        } else {
+            combinedIntradayHeartRateDataProvider(
+                props.dataSources,
+                intervalStart,
+                intervalEnd,
+                props.aggregationOption,
+                props.aggregationIntervalMinutes
+            ).then((data: IntradayHeartRateData) => {
+                transformToChartData(data);
+            }).catch((error: any) => {
+                console.log("error");
+                console.log(error);
+            });
+        }
+    }
+
+    function transformToChartData(data: IntradayHeartRateData) {
+        const start = intervalStart;
+        const end = intervalEnd;
+        const interval: Interval = { start, end };
+        const options: any = { step: 5 };
+        var expectedIntervals = eachMinuteOfInterval(interval, options);
+        var chartData: any[] = expectedIntervals.map((date) => transformToDataPoint(date, data));
+
+        var yDomain: number[] = [];
+        Object.values(chartData).forEach(dataPoint => {
+            dataPoint.value !== undefined && yDomain.push(dataPoint.value);
+        });
+
+        const yMaxValue = Math.max(...yDomain);
+        const yMinValue = Math.min(...yDomain);
+
+        setOptions({ domainMin: yMinValue, domainMax: yMaxValue, thresholds: props.thresholds, connectNulls: false });
+        setData(chartData);
+        setLoading(false);
+    }
+
+    function transformToDataPoint(date: Date, data: IntradayHeartRateData) {
+        var dayKey = date.getTime();
+        var intradayDataPoint: IntradayDataPoint = { timestamp: dayKey, date: date };
+        if (data[dayKey]) {
+            intradayDataPoint.value = (data[dayKey]).statistics[`${props.aggregationOption}`];
+            intradayDataPoint.rawValue = intradayDataPoint.value;
+        }
+
+        return intradayDataPoint;
+    }
+
+    const IntradayHrToolTip = ({ active, payload, label }: any) => {
+        if (active && payload && payload.length) {
+            return (
+                <div className="mdhui-time-series-tooltip">
+                    <div className="mdhui-single-value-tooltip-value">{`${Math.round(payload[0].value)} bpm`}</div>
+                    <div className="mdhui-time-series-tooltip-date">{format(payload[0].payload.date, "h:mm aaa")}</div>
+                </div>
+            );
+        }
+        return null;
+    };
+
+    useInitializeView(initialize, [], [props.previewState, props.dataSources, dateRangeContext?.intervalStart]);
+
+    return <div ref={props.innerRef}>
+        {loading && <LoadingIndicator></LoadingIndicator>}
+        {!loading &&
+            <>
+                {data && <TimeSeriesChart intervalType='Day' intervalStart={intervalStart} data={data}
+                    chartHasData={(!loading && isNumber(options?.domainMax) && options?.domainMax > 0)} 
```

Comment:
I think this can simplify to: `chartHasData={!loading && !!options?.domainMax && options.domainMax > 0}`.

My IDE is complaining about `options?.domainMax` possibly being `undefined` at the moment.